### PR TITLE
Feature/heroku

### PR DESCRIPTION
**Pull request by JuanitoFatas**
This Pull Request
- Install rails_12factor for staging and production
- Specify ruby version in Gemfile
- Update README.md ([View](https://github.com/rubytaiwan/rubytw-reboot/blob/254f1c3b13162d41ce7f005da4fc258af7d007c4/README.md))
  - Add link to production app
  - Link to deployment section
- Creates DEPLOYMENT.md ([View](https://github.com/rubytaiwan/rubytw-reboot/blob/4c54becf007ca3047b83abe9e8062779bd56a467/DEPLOYMENT.md))
- Follow [getting started guide from heroku devcenter](https://devcenter.heroku.com/articles/getting-started-with-rails4) to setup staging and production apps on Heroku
- staging config in `config/secrets.yml`

Implements #4

/cc @rubytaiwan/rubyists

### CODE REVIEW

**Comment by AnNOtis** [comment]
:+1:

File: config/environments/staging.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  # config.assets.css_compressor = :sass
+
+  # Do not fallback to assets pipeline if a precompiled asset is missed.
+  config.assets.compile = false
+
+  # Asset digests allow you to set far-future HTTP expiration dates on all assets,
+  # yet still be able to expire them through the digest params.
+  config.assets.digest = true
+
+  # `config.assets.precompile` and `config.assets.version` have moved to config/initializers/assets.rb
```
Comment (JuanitoFatas) [inline config/environments/staging.rb:38]: Ignore it since this is Rails generated file.

**Comment by sunkai612** [comment]
link of [DEPLOYMENT.md](https://github.com/rubytaiwan/rubytw-reboot/blob/254f1c3b13162d41ce7f005da4fc258af7d007c4/DEPLOYMENT.md) on `README.md` didn't work

seems like it should be:
https://github.com/rubytaiwan/rubytw-reboot/blob/4c54becf007ca3047b83abe9e8062779bd56a467/DEPLOYMENT.md

**Comment by sunkai612** [comment]
Oops...
I found that the link is not working becuase it's not merged yet :stuck_out_tongue: 

looks like it's ready for merge once CI checking is done :+1:

**Comment by sinorga** [comment]
It looks CI failed at ruby version (2.1.7) which is not consistent with the version (2.2.3) specified in Gemfile. So maybe we need to remove the line in Gemfile for passing the test of difference runtime environment? or any other good suggestions :smiley:

**Comment by JuanitoFatas** [comment]
We can have multiple Gemfiles and travis CI supports this. But let's keep it simple to only support the Ruby version we are using for now. Previously asking to test multiple Ruby versions was just an exercise for @puff-tw :trollface:

**Pull request merged**